base_layer/wallet/src/output_manager_service/storage/sqlite_db/output_sql.rs

```diff
@@ -102,6 +104,20 @@ impl OutputSql {
         Ok(outputs::table.filter(outputs::status.eq(status as i32)).load(conn)?)
     }
 
+    /// Retrieves UTXOs than can be spent, sorted by priority, then value from smallest to largest.
+    pub fn fetch_unspent_outputs_for_spending(
+        tip_height: i64,
+        conn: &SqliteConnection,
+    ) -> Result<Vec<OutputSql>, OutputManagerStorageError> {
+        Ok(outputs::table
+            .filter(outputs::status.eq(OutputStatus::Unspent as i32))
+            .filter(outputs::script_lock_height.le(tip_height))
+            .filter(outputs::maturity.le(tip_height))
+            .order(outputs::spending_priority.asc())
+            .order(outputs::value.asc())
```

Comment:
This must depend on the UTXOSelectionStrategy. There is a strategy that chooses Largest first. So would take that param into this call and adjust this query based on the strategy selected.